src/wallet/scriptpubkeyman.cpp

```diff
@@ -867,6 +867,33 @@ bool LegacyScriptPubKeyMan::RemoveWatchOnly(const CScript &dest)
     return true;
 }
 
+bool LegacyScriptPubKeyMan::PurgeWatchOnly(const CScript& dest)
+{
+    {
+        LOCK(cs_KeyStore);
+        setWatchOnly.erase(dest);
+        CPubKey pubKey;
+        if (ExtractPubKey(dest, pubKey)) {
+            CKeyID key_id = pubKey.GetID();
+            mapWatchKeys.erase(key_id);
+            mapKeyMetadata.erase(key_id);
+            if (pubKey.IsCompressed()) {
+                CScript script = GetScriptForDestination(WitnessV0KeyHash(key_id));
+                CScriptID id(script);
+                mapScripts.erase(id);
+                m_script_metadata.erase(id);
+            }
+        }
+    }
+
+    if (!HaveWatchOnly())
+        NotifyWatchonlyChanged(false);
+    if (!WalletBatch(m_storage.GetDatabase()).EraseWatchOnly(dest))
+        return false;
```

Comment:
Please see doc/developer-notes.md about formatting conditionals
```
  - If an `if` only has a single-statement `then`-clause, it can appear
    on the same line as the `if`, without braces. In every other case,
    braces are required, and the `then` and `else` clauses must appear
    correctly indented on a new line.
```